src/interpolation_transfer.jl

```diff
@@ -168,6 +165,79 @@ function transfer_blochwave_kpt(ψk_in, basis_in::PlaneWaveBasis{T}, kpt_in::Kpo
     ψk_out
 end
 
+"""
+Transfer an array `ψk_in` expanded on `kpt_in`, and produce ``ψ(r) e^{i ΔG·r}`` expanded on
+`kpt_out`.
+Useful for phonons.
+"""
+function transfer_blochwave_kpt(ψk_in, basis::PlaneWaveBasis, kpt_in, kpt_out, ΔG)
+    ψk_out = zeros(eltype(ψk_in), length(G_vectors(basis, kpt_out)), size(ψk_in, 2))
+    for (iG, G) in enumerate(G_vectors(basis, kpt_in))
+        # e^i(kpt_in + G)r = e^i(kpt_out + G')r, where
+        # kpt_out + G' = kpt_in + G = kpt_out + ΔG + G, and
+        # G' = G + ΔG
+        idx_Gp_in_kpoint = index_G_vectors(basis, kpt_out, G - ΔG)
+        if !isnothing(idx_Gp_in_kpoint)
+            ψk_out[idx_Gp_in_kpoint, :] = ψk_in[iG, :]
+        end
+    end
+    ψk_out
+end
+
+"""
+Real-space version of the previous function.
+"""
+function transfer_blochwave_kpt_real(ψk_in, basis::PlaneWaveBasis, kpt_in::Kpoint,
+                                    kpt_out::Kpoint, ΔG; q=nothing)
+    ψk_out = zeros(eltype(ψk_in), length(kpt_out.G_vectors), size(ψk_in, 2))
+    exp_ΔG = cis2pi.(-dot.(Ref(ΔG), r_vectors(basis)))
+    for n in 1:size(ψk_in, 2)
+        ψk_out[:, n] = fft(basis, kpt_out, exp_ΔG .* ifft(basis, kpt_in, ψk_in[:, n]))
+    end
+    ψk_out
+end
+
+"""
+For x ∈ ℝ, return the reduced coordinate in [-½, ½).
+"""
+function mod_half(x::T; tol=sqrt(eps(T))) where {T}
+    y =  mod(x, 1)
+    y ≥ 0.5 - tol && return y - 1
+    return y
+end
+
+"""
+Find the equivalent of `k`-point `kpt` ∈ ℝ³ and its position `index` in a list `kpoints`
+∈ [-½, ½).
+`ΔG` is the vector of ℤ³ such that `kpt_out = kpt + ΔG`
+"""
+function find_equivalent_kpt(kpoints::Vector{Kpoint{T}}, kpt; tol=100*eps(T)) where {T}
+    coordinates = getfield.(kpoints, :coordinate)
+
+    kpt_normalized_coordinate = mod_half.(kpt.coordinate)
+    index = findall(e -> isapprox(e, kpt_normalized_coordinate, atol=tol), coordinates)
```

Comment:
We have `is_approx_integer` in SymOp, you could move it to common/ and use it here

Reply:
(and get rid of mod_half)